presto-hive-metastore/src/main/java/com/facebook/presto/hive/RetryDriver.java

```diff
@@ -139,6 +139,11 @@ public <V> V run(String callableName, Callable<V> callable)
                 return callable.call();
             }
             catch (Exception e) {
+                // Immediately stop retry attempts once an interrupt has been received
+                if (e instanceof InterruptedException || Thread.currentThread().isInterrupted()) {
```

Comment:
Sorry for the late comment, Checking the isInterrupted will clear the interrupt flag on the thread, is that expected ? I am not sure who is responsible for clearing the interrupt flag in the Presto codebase.

Sleep on line 163, tries to  rightly set the interrupted flag.

Reply:
My bad, isInterrupted does not clear the flag, so it is good. but on handling the interruptedException, there are still differences.

Reply:
We’re rethrowing InterruptedException (or whatever other exception was thrown if the current thread is interrupted) but not “handing” it other than to ensure that we stop retrying and let the exception propagate.

Reply:
Thanks this makes sense